Amethyst/Layout/ThreeColumnLayout.swift

```diff
@@ -0,0 +1,237 @@
+}
+
+final class ThreeColumnReflowOperation: ReflowOperation {
+    private let layout: ThreeColumnLayout
+    private let mainIs: WhichColumn
+
+    init(screen: NSScreen, windows: [SIWindow], layout: ThreeColumnLayout, frameAssigner: FrameAssigner, mainIs: WhichColumn) {
+        self.layout = layout
+        self.mainIs = mainIs
+        super.init(screen: screen, windows: windows, frameAssigner: frameAssigner)
+    }
+
+    func frameAssignments() -> [FrameAssignment] {
+        guard !windows.isEmpty else {
+            return []
+        }
+
+        let mainPaneCount: UInt = UInt(min(windows.count, layout.mainPaneCount))
+        let nonMainCount: UInt = UInt(windows.count) - mainPaneCount
+        let secondaryPaneCount: UInt = nonMainCount >> 1
+        let tertiaryPaneCount: UInt = nonMainCount - secondaryPaneCount
+
+        let hasSecondaryPane = secondaryPaneCount > 0
+        let hasTertiaryPane = tertiaryPaneCount > 0
+
+        let screenFrame = screen.adjustedFrame()
+
+        let mainPaneWindowHeight = round(screenFrame.height / CGFloat(mainPaneCount))
+        let secondaryPaneWindowHeight = hasSecondaryPane ? round(screenFrame.height / CGFloat(secondaryPaneCount)) : 0.0
+        let tertiaryPaneWindowHeight = hasTertiaryPane ? round(screenFrame.height / CGFloat(tertiaryPaneCount)) : 0.0
+
+        var mainPaneWindowWidth: CGFloat = 0.0
+        var secondaryPaneWindowWidth: CGFloat = 0.0
+        if hasSecondaryPane && hasTertiaryPane {
+            mainPaneWindowWidth = round(screenFrame.width * CGFloat(layout.mainPaneRatio))
+            secondaryPaneWindowWidth = round((screenFrame.width - mainPaneWindowWidth) / 2)
+        } else if hasSecondaryPane {
+            secondaryPaneWindowWidth = round(screenFrame.width * CGFloat(layout.mainPaneRatio))
+            mainPaneWindowWidth = screenFrame.width - secondaryPaneWindowWidth
+        } else {
+            mainPaneWindowWidth = screenFrame.width
+        }
+
+        let tertiaryPaneWindowWidth = screenFrame.width - mainPaneWindowWidth - secondaryPaneWindowWidth
+
+        let focusedWindow = SIWindow.focused()
+
+        return windows.reduce([]) { frameAssignments, window -> [FrameAssignment] in
+            var assignments = frameAssignments
+            var windowFrame = CGRect.zero
+            let windowIndex: UInt = UInt(frameAssignments.count)
+
+            let whichPane: WhichPane = {
+                // main windows are indexes [0, mainPaneCount)
+                if windowIndex < mainPaneCount { return WhichPane.main }
+                // secondary windows are indexes [mainPaneCount, mainPaneCount + secondaryPaneCount)
+                // tertiary windows are indexes [mainPaneCount + secondaryPaneCount, ...)
+                else if windowIndex >= (mainPaneCount + secondaryPaneCount) { return WhichPane.tertiary }
+                else { return WhichPane.secondary }
+            }()
+
+            // index within the column
+            let subIndex: UInt = {
```

Comment:
I went with `paneIndex` since the new function `getCoordsForWindowIndex` then can return a `Pane` and a `paneIndex`, which seems clear/elegant to me, but I'm not strongly attached and can switch back to `columnIndex` if you prefer.